crates/cli/src/lib.rs

```diff
@@ -559,7 +570,29 @@ pub fn build(
                         DEFAULT_ROC_FILENAME
                     )
                 }
-                _ => eprintln!("\nThis file was not found: {}\n\nYou can run `roc help` for more information on how to provide a .roc file.\n", expected_file_path_string),
+                _ => {
+                    let mut error_lines = Vec::new();
+                    error_lines.push(format!(
+                        "This file was not found: {}",
+                        expected_file_path_string
+                    ));
+                    // Add some additional hints if run as `roc [FILENAME]`.
+                    if matches.subcommand().is_none() {
+                        if let Some(possible_typo) = path.to_str() {
+                            if let Some((nearest_command, _)) =
```

Comment:
nit: We may want to skip this message if the possible_typo ends in `.roc`.

Also, even more optional, if the distance is large, maybe we should skip the message as well.

Reply:
Good ideas! I added the `.roc` check, but thought I'd be a little more permissive to start when it came to subcommand matching. I included the distance in the return value when I wrote the matching function so it's easy to add additional restrictions later if there are too many false positives. Happy to add it now though if you prefer.